torch/csrc/utils/tensor_flatten.cpp

```diff
@@ -54,7 +54,7 @@ std::vector<TensorGroup> take_tensors(
   // End case. Look for any remaining groups and return them.
   for (auto& entry : groups) {
     auto& group = entry.second;
-    if (!fine_grained && group.size == 0) {
+    if (group.tensors.empty()) {
```

Comment:
The non-find-grained version outputs result groups by doing
```cpp
results.emplace_back();
std::swap(results.back(), type_group);
```

Meaning there can be default constructed tensor groups in the `groups` map.
`groups.size == 0` was likely meant to exclude these default constructed groups but it also removed groups with 0-size tensors as well.

Reply:
OK LGTM